src/cameras.jl

```diff
@@ -1,160 +1,189 @@
-abstract type Camera end
-camera_set!(::RenderModel, cam::Camera, ::EntityFrame{S,D,I}, ::R, canvas_width::Int, canvas_height::Int) where {S,D,I,R} = error("camera_set! not implemented for Camera $cam")
+"""
+    CameraState
+Representation of camera parameters such as position, rotation and zoom level.
 
-mutable struct StaticCamera <: Camera
-    pos::VecE2
-    zoom::Float64 # [pix/meter]
-    StaticCamera(pos::VecE2, zoom::Float64=3.0) = new(pos, zoom)
-    StaticCamera(x::Float64, y::Float64, zoom::Float64=3.0) = new(VecE2(x,y), zoom)
+ - `camera_center::VecE2{Real}`: position of camera in [N,E] relative to the mean point. meters
+ - `camera_zoom::Real`: camera zoom in [pix/m]
+ - `camera_rotation::Real`: camera rotation in [rad]
+"""
+@with_kw mutable struct CameraState
+    position  :: VecE2 = VecE2(0.,0.)
+    zoom      :: Real = 1.
+    rotation  :: Real = 0.
+    canvas_width  :: Int64 = DEFAULT_CANVAS_WIDTH
+    canvas_height :: Int64 = DEFAULT_CANVAS_HEIGHT
 end
-function camera_set!(rendermodel::RenderModel, cam::StaticCamera, canvas_width::Int, canvas_height::Int)
+position(cs::CameraState) = cs.position
+zoom(cs::CameraState) = cs.zoom
+rotation(cs::CameraState) = cs.rotation
+canvas_width(cs::CameraState) = cs.canvas_width
+canvas_height(cs::CameraState) = cs.canvas_height
+
+camera_move!(cs::CameraState, dx::Real, dy::Real) = cs.position = cs.position + VecE2(dx, dy)
+camera_move!(cs::CameraState, Δ::VecE2) = cs.position = cs.position + Δ
+camera_move_pix!(cs::CameraState, dx::Real, dy::Real) = cs.position = cs.position + VecE2(dx/cs.zoom, dy/cs.zoom)
+camera_move_pix!(cs::CameraState, Δ::VecE2) = cs.position = cs.position + VecE2(Δ.x/cs.zoom, Δ.y/cs.zoom)
+camera_rotate!(cs::CameraState, θ::Real) = cs.rotation += θ # [radians]
+camera_zoom!(cs::CameraState, factor::Real) = cs.zoom *= factor
```

Comment:
@mattuntergassmair these functions do not seem to be used anywhere, what is the use case?

Reply:
That's correct, I don't think they are being used. I did not introduce them, it's legacy code coming from `rendermodels.jl` and it looked useful enough to keep it around - but we can also get rid of it.